R/rules-spaces.R

```diff
@@ -150,7 +152,8 @@ remove_space_after_opening_paren <- function(pd_flat) {
 }
 
 remove_space_before_closing_paren <- function(pd_flat) {
-  paren_after <- pd_flat$token %in% c("')'", "']'")
+  closing_braces <- c("')'", "']'")
```

Comment:
@lorenzwalthert Is it deliberate to not include `"RBB"` here?

Reply:
What is `"RBB"` again?

Reply:
There is no RBB (corresponding to LBB); I was hallucinating about having seen it 🙈

Reply:
For whatever reason, `[[` is LBB, but `]]` is two right brackets 🤷 

``` r
styler:::get_parse_data("x[[1]]")
#>   line1 col1 line2 col2 id parent     token terminal   text pos_id short
#> 1     1    1     1    6 11      0      expr    FALSE x[[1]]      1 x[[1]
#> 2     1    1     1    1  1      3    SYMBOL     TRUE      x      2     x
#> 3     1    1     1    1  3     11      expr    FALSE      x      3     x
#> 4     1    2     1    3  2     11       LBB     TRUE     [[      4    [[
#> 5     1    4     1    4  4      5 NUM_CONST     TRUE      1      5     1
#> 6     1    4     1    4  5     11      expr    FALSE      1      6     1
#> 7     1    5     1    5  6     11       ']'     TRUE      ]      7     ]
#> 8     1    6     1    6 10     11       ']'     TRUE      ]      8     ]
```

<sup>Created on 2024-05-11 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
Sounds like a case for the parser crew to fix...